hierarchicalforecast/methods.py

```diff
@@ -837,6 +837,23 @@ def _get_PW_matrices(self,
                 # Protection: final ridge diagonal protection
                 W = (lmd * tar + (1 - lmd) * covm) + self.mint_shr_ridge
 
+                UtW = Ut @ W
+            elif self.method == 'mint_shrink':
+                # Compute nans
+                nan_mask = np.isnan(residuals.T)
+                # Compute shrunk empirical covariance
+                if np.any(nan_mask):
+                    residuals_mean = np.nanmean(residuals, axis=0, dtype=np.float64, keepdims=True).T
+                    residuals_std = np.maximum(np.nanstd(residuals, axis=0, dtype=np.float64, keepdims=True), 1e-6).T
+                    safe_residuals = np.nan_to_num(residuals.T, copy=False)
+                    Xs = (safe_residuals - residuals_mean) / (residuals_std + 2e-8)
```

Comment:
The only issue I can see with this is that the samples of observations for each pair of time series may differ if the masks do not align, so the means would have to be calculated in the loop for each $i$ and $j$ before calculating the deviations and standardising. I know that unfortunately this would slow down the NaN version.

Reply:
True, it needs to be calculated in-loop per pair, fixed that, thanks.